src/parser/index.ts

```diff
@@ -214,3 +228,48 @@ function extractTokens(ctx: Context) {
     return /^[^\w$]$/iu.test(c);
   }
 }
+
+/**
+ * Extracts style source from a SvelteStyleElement and parses it into a PostCSS AST.
+ */
+function parseStyleAst(
+  styleElement: SvelteStyleElement,
+  ctx: Context
+): Root | null {
+  if (!styleElement.endTag) {
+    return null;
+  }
+  let lang = "css";
+  for (const attribute of styleElement.startTag.attributes) {
+    if (
+      attribute.type === "SvelteAttribute" &&
+      attribute.key.name === "lang" &&
+      attribute.value.length > 0 &&
+      attribute.value[0].type === "SvelteLiteral"
+    ) {
+      lang = attribute.value[0].value;
+    }
+  }
+  let parseFn: Parser<Root> | undefined = postcss.parse;
+  switch (lang) {
+    case "css":
+      parseFn = postcss.parse;
+      break;
+    case "scss":
+      parseFn = SCSSparse;
+      break;
+    default:
+      console.warn(`Unknown <style> block language "${lang}".`);
+      parseFn = undefined;
+  }
+  if (parseFn !== undefined) {
+    const styleCode = ctx.code.slice(
+      styleElement.startTag.range[1],
+      styleElement.endTag.range[0]
+    );
+    return parseFn(styleCode, {
```

Comment:
In case of CSS parse errors, I find it useful to return that information. You can add a new rule to display parse errors as warnings. Rules can also optionally use `postcss-safe-parser` to fix parsing errors.
https://www.npmjs.com/package/postcss-safe-parser

Reply:
(However, I think using stylelint makes more sense than creating a new CSS parse error rule.)

Reply:
You mean errors produced by SCSS? Hmm, I am not sure what would happen currently, maybe it would all crash? I'll investigate more...

Reply:
I'm talking about CSS syntax errors.
I don't think we should stop building the entire AST just because svelte file has CSS with syntax errors. I think we need to handle syntax errors with `try {} catch {}`.
I think it would be nice to add to the information in getStyleContext if there were any syntax errors.

Reply:
I'd like to add the parse errors, however, I am currently blocked by postcss/postcss#1839 :(

Reply:
Well, see the comments in the linked postcss issue - IMHO it's not worth the added complexity to list parse errors.

Reply:
I think that it is enough if we can get one error information :wink: